flake8_type_checking/checker.py

```diff
@@ -485,6 +485,41 @@ def is_type_checking(self, node: ast.AST) -> bool:
             or (self.type_checking_alias is not None and hasattr(node, 'id') and (node.id == self.type_checking_alias))
         )
 
+    def is_type_checking_true(self, node: ast.Compare) -> bool:
+        """Determine whether `if TYPE_CHECKING is True` statement is passed."""
+        # Check for `left` attribute in the node.
+        has_left_node = hasattr(node, 'left') is True
+
+        # Check for `node.id`
+        has_left_id = (
+            has_left_node
+            and hasattr(node.left, 'id') is True
+            and (
+                node.left.id == 'TYPE_CHECKING'
+                or (self.type_checking_alias is not None and node.left.id == self.type_checking_alias)
+            )
+        )
+
+        # Check for `node.attr`
+        has_left_attr = has_left_node and hasattr(node.left, 'attr') is True and node.left.attr == 'TYPE_CHECKING'
+
+        # Generate a final boolean value for the left side of expression.
+        type_checking_on_left = has_left_id or has_left_attr
+
+        # Operator should be `is`.
+        operator_is = hasattr(node, 'ops') is True and len(node.ops) == 1 and isinstance(node.ops[0], ast.Is) is True
+
+        # Right side of the operator should be a constant `True`.
+        right_value_true = (
+            hasattr(node, 'comparators') is True
+            and len(node.comparators) == 1
+            and isinstance(node.comparators[0], ast.Constant) is True
+            and hasattr(node.comparators[0], 'value') is True
+            and node.comparators[0].value is True
+        )
+
+        return type_checking_on_left and operator_is and right_value_true
```

Comment:
Very nice work :+1: 

I think there's a few conditions here we can leave out, like the `hasattr(..., 'value')` for `ast.Constant` and similar checks. These will always hold I believe. Taking that into consideration, and re-using the `is_type_checking` method from above this, I believe we might be able to reduce the code to this:

```suggestion
def is_type_checking_true(self, node: ast.Compare) -> bool:
    """
    An ast.Compare node has a `left`, `ops`, and `comparators` attribute.

    Here we want to check whether our node corresponds to
    
        `if TYPE_CHECKING is True`
                ^         ^    ^____ comparators
       left ____|       ops
    """
    # Left side should be a TYPE_CHECKING block
    if not hasattr(node, 'left') or not self.is_type_checking(node.left):
        return False
    
    # Operator should be `is`
    if not len(node.ops) == 1 or not isinstance(node.ops[0], ast.Is):
        return False
    
    # Right side should be `True`
    if not len(node.comparators) == 1 or not isinstance(node.comparators[0], ast.Constant) or not node.comparators[0].value is True:
        return False

    return True
```